src/rulesets/Base/base.jl

```diff
@@ -119,6 +119,40 @@ function rrule(::typeof(hypot), z::Complex)
     return (Ω, hypot_pullback)
 end
 
+# Note that `hypot` with two arguments has rules in `fastmath_able.jl`
+
+function frule(
+    (_, Δx, Δy, Δz, Δxs...),
+    ::typeof(hypot),
+    x::T,
+    y::T,
+    z::T,
+    xs::Vararg{T, N}
+) where {T<:Union{Real,Complex}, N}
+    Ω = hypot(x, y, z, xs...)
+    n = ifelse(iszero(Ω), one(Ω), Ω)
+    ∂Ωxyz = realdot(x, Δx) + realdot(y, Δy) + realdot(z, Δz)
+    ∂Ωxs = sum(realdot(xi, Δxi) for (xi, Δxi) in zip(xs, Δxs); init=zero(∂Ωxyz))
+    ∂Ω = (∂Ωxyz + ∂Ωxs) / n
```

Comment:
Arguably a bit simpler:
```suggestion
    ∂Ω = sum(map(realdot, (x, y, z, xs...), (Δx, Δy, Δz, Δxs...))) / n
```